<launch>
  <arg name="path_frame" default ="base_link_p"/>
  <arg name="odom_topic" default ="/gazebo/forklift_controllers/odom"/>
  <arg name="cmd_vel_topic" default ="/cmd_vel_raw"/>

 <node pkg="pallet_docking_xsquare" type="docking_control" name="docking_control" output = "screen" >
   <param name="path_frame" value="$(arg path_frame)"/>
    
    <!-- Pure Pursuit parameters -->
   <param name="look_ahead_time" value="3.6"/>      
   <param name="min_look_ahead_dis" value="0.1"/>  
   <param name="max_look_ahead_dis" value="2.5"/> 
   <param name="max_steering" value="1.57"/>     
   <param name="min_steering" value="-1.57"/>  
   <param name="kp" value="2.0"/>     
   <param name="ki" value="0.1"/> 
   <param name="goal_correct_yaw" value="0.1"/>  
   <param name="max_angular_vel" value="1.5"/>  <!-- change from the stable branch -->
   <param name="use_point_interpolate" value="true"/> 

   <!-- Fuzzy controller parameters -->
   <param name="fuzzy_lookahead_dis" value="1.0"/>      <!-- ZERO value for output velocity --> 

   <param name="min_vel" value="0.03"/>      
   <param name="output_VS" value="0.05"/>    
   <param name="output_S" value="0.1"/>     
   <param name="output_M" value="0.2"/>     
   <param name="max_vel" value="0.3"/>       

   <!-- <param name="min_vel" value="0.03"/>      
   <param name="output_VS" value="0.05"/>    
   <param name="output_S" value="0.2"/>     
   <param name="output_M" value="0.35"/>      
   <param name="max_vel" value="0.5"/>        -->

   <param name="min_dis" value="0.2"/>      <!-- SMALL distance for input -->
   <param name="goal_M_b" value="0.35"/>      <!-- MEDIUM distance for input -->
   <param name="max_dis" value="0.5"/>      <!-- BIG distance for input -->

   <param name="min_steer" value="-1.495"/>  <!-- NEGATIVE BIG steering for input -->
   <param name="steer_NS_b" value="-0.8"/>   <!-- NEGATIVE SMALL steering for input -->
   <param name="steer_Z_b" value="0.0"/>     <!-- ZERO steering for input -->
   <param name="steer_PS_b" value="0.8"/>    <!-- POSITIVE SMALL steeringfor input -->
   <param name="max_steer" value="1.495"/>   <!-- POSITIVE BIG steering for input -->

   <param name="max_abs_vel_change" value="0.01"/>   <!-- dt = 0.02, a = max_abs_vel_change/dt-->
 
  <remap from="/cmd_vel" to = "$(arg cmd_vel_topic)"/>
  <remap from="/gazebo/forklift_controllers/odom" to = "$(arg odom_topic)"/>

 </node>
</launch>
